### PR TITLE
fix: Unify custom model logits extraction across all inference methods

### PR DESCRIPTION
**Pull request by zpqiu**
This commit fixes incomplete handling of custom MoE models (e.g., from AutoModel) that output logits directly as tensors instead of objects with a .logits attribute.

Changes:
- get_logprobs(): Updated to use complete logits extraction logic with proper handling for all three cases (tensor output, missing .logits, has .logits)
- score(): Added missing isinstance check for tensor outputs and fixed the float32 conversion bug that incorrectly referenced outputs.logits
- get_topk_logits(): Already had the correct logic, ensuring consistency

All methods now follow the same pattern as the train() method for extracting logits from model outputs, preventing AttributeError when using custom models.

Related to issue #1810

# What does this PR do ?

**Add a one line overview of what this PR aims to accomplish.**

# Issues
List issues that this PR closes ([syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)):

# Usage
* **You can potentially add a usage example below**

```python
# Add a code snippet demonstrating how to use this
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/NVIDIA-NeMo/RL/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you run the unit tests and functional tests locally? Visit our [Testing Guide](/NVIDIA-NeMo/RL/blob/main/docs/testing.md) for how to run tests
- [ ] Did you add or update any necessary documentation? Visit our [Document Development Guide](/NVIDIA-NeMo/RL/blob/main/docs/documentation.md) for how to write, build and test the docs.

# Additional Information
* ...

## Summary by CodeRabbit

* **Refactor**
  * Improved model output handling for better compatibility across different architectures.
  * Optimized memory management for enhanced efficiency.
  * Ensured consistent data type handling across processing operations.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py [hunk 1]
```diff
@@ -823,7 +823,15 @@ def get_logprobs(
...
-                    logits = outputs.logits if hasattr(outputs, "logits") else outputs
+                    # Get logits
+                    if isinstance(outputs, (torch.Tensor, DTensor)):
+                        # custom models (e.g., those coming from AutoModel) can output logits directly
+                        logits = outputs
+                    elif not hasattr(outputs, "logits"):
+                        logits = self.model.lm_head(outputs.last_hidden_state)
+                    else:
+                        logits = outputs.logits
+                    del outputs
```
[the remark this answers is not in the document]
Comment (zpqiu) [reply nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py]: done

**Comment by terrykong** [comment]
Yes, I believe so. @hemildesai can you make sure your PR reflects?
